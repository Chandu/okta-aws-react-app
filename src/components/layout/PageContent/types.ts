import { WithThemeProp } from '@utils';
import { ReactElement } from 'react';

export interface Props extends WithThemeProp {
    pageTitle?: string | ReactElement<any>;
}
