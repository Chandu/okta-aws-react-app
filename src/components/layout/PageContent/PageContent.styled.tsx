import { styled } from '@utils';

export const PageContent = styled('main')`
    overflow: auto;
    border-radius: 0.8rem;
    background-color: ${({ theme }) => theme.colors.grey[50]};
`;

export const PageHeaderContentWrapper = styled('div')`
    padding: 4rem;
    background-color: ${({ theme }) => theme.colors.white};
    margin-bottom: 0.8rem;
    border-radius: 0.8rem;
`;

export const ChildrenContentWrapper = styled('div')`
    padding: 1rem 4rem;
    background-color: white;
    border-radius: 0.8rem;
`;
