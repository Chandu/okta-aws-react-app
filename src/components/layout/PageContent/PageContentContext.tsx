import React from 'react';
export const PageContentContext = React.createContext(
    {} as {
        pageContentRef: React.MutableRefObject<HTMLElement | null>;
    }
);
