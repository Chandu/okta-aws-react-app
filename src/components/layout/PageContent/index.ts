export { PageContent as default } from './PageContent';
export * from './PageContentContext';
