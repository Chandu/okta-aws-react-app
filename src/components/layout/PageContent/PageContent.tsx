import { Text } from '@medly-components/core';
import { isValidStringOrNumber, WithStyle } from '@medly-components/utils';
import { FC } from 'react';
import * as Styled from './PageContent.styled';
import { Props } from './types';

export const PageContent: FC<Props> & WithStyle = props => {
    const { pageTitle, children } = props;
    return (
        <Styled.PageContent>
            <Styled.PageHeaderContentWrapper>
                {pageTitle && isValidStringOrNumber(pageTitle) ? (
                    <Text textVariant="h2" id="page-title">
                        {pageTitle}
                    </Text>
                ) : (
                    pageTitle
                )}
            </Styled.PageHeaderContentWrapper>

            {children && <Styled.ChildrenContentWrapper>{children}</Styled.ChildrenContentWrapper>}
        </Styled.PageContent>
    );
};

PageContent.displayName = 'PageContent';
PageContent.Style = Styled.PageContent;

export default PageContent;
