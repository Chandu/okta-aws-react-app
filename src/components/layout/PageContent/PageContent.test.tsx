import { renderWithRouter } from '@test-utils';
import PageContent from './PageContent';

describe('PageContent component', () => {
    it('should render properly', () => {
        const { container } = renderWithRouter(<PageContent pageTitle="test">Demo PageContent</PageContent>);
        expect(container).toMatchSnapshot();
    });
});
