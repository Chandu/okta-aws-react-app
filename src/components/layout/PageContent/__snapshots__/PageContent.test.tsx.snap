// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PageContent component should render properly 1`] = `
.c2 {
  margin: 0;
  color: inherit;
  font-size: 3.2rem;
  font-weight: 500;
  -webkit-letter-spacing: -0.04rem;
  -moz-letter-spacing: -0.04rem;
  -ms-letter-spacing: -0.04rem;
  letter-spacing: -0.04rem;
  line-height: 4.0rem;
  text-align: initial;
}

.c0 {
  overflow: auto;
  border-radius: 0.8rem;
  background-color: #F7F8F9;
}

.c1 {
  padding: 4rem;
  background-color: #ffffff;
  margin-bottom: 0.8rem;
  border-radius: 0.8rem;
}

.c3 {
  padding: 1rem 4rem;
  background-color: white;
  border-radius: 0.8rem;
}

<div>
  <main
    class="c0"
  >
    <div
      class="c1"
    >
      <h2
        class="c2"
        id="page-title"
      >
        test
      </h2>
    </div>
    <div
      class="c3"
    >
      Demo PageContent
    </div>
  </main>
</div>
`;
